packages/shared/src/components/plus/PlusInfo.tsx

```diff
@@ -51,105 +87,80 @@ export const PlusInfo = ({
         className="mb-6"
       >
-        Upgrade to daily.dev Plus for an enhanced, ad-free experience with
-        exclusive features and perks to level up your game.
+        {description}
       </Typography>
-      <Typography
-        tag={TypographyTag.P}
-        type={TypographyType.Callout}
-        color={TypographyColor.Tertiary}
-        className="mb-4"
-        bold
+      <ConditionalWrapper
+        condition={!giftingUser}
+        wrapper={(component) => (
+          <div className="mb-4 flex flex-row items-center justify-between">
+            <span>{component}</span>
+            <Button
+              icon={<GiftIcon />}
+              size={ButtonSize.XSmall}
+              variant={ButtonVariant.Float}
+              onClick={() =>
+                openModal({
+                  type: LazyModal.GiftPlus,
+                  props: { onSubmit: onUserChange },
+                })
+              }
+            >
+              Buy as a gift
+            </Button>
+          </div>
+        )}
       >
-        Billing cycle
-      </Typography>
+        <Typography
+          tag={TypographyTag.P}
+          type={TypographyType.Callout}
+          color={TypographyColor.Tertiary}
+          className={giftingUser && 'mb-4'}
+          bold
+        >
+          {subtitle}
+        </Typography>
+      </ConditionalWrapper>
+      {!!giftingUser && (
+        <GiftingSelectedUser
+          user={giftingUser}
+          onClose={() => onUserChange(null)}
+        />
+      )}
       <div className="min-h-[6.125rem] rounded-10 border border-border-subtlest-tertiary">
-        {productOptions.map((option) => {
-          const { label, value, price, currencyCode, extraLabel } = option;
-          const checked = selectedOption === value;
-          const isEarlyAccess = value === earlyAdopterPlanId;
-          return (
-            <RadioItem
-              key={label}
-              name={label}
-              id={`${label}-${value}`}
-              value={value}
-              checked={checked}
-              onChange={() => {
-                onChange(value);
-                logSubscriptionEvent({
-                  event_name: LogEvent.SelectBillingCycle,
-                  target_id: label.toLowerCase(),
-                });
-              }}
-              className={{
-                content: classNames(
-                  'min-h-12 rounded-10 !p-2',
-                  checked
-                    ? '-m-px border border-border-subtlest-primary bg-surface-float'
-                    : undefined,
-                ),
-              }}
-            >
-              <div className="flex flex-1 flex-wrap items-center gap-x-3 gap-y-1">
-                <Typography
-                  tag={TypographyTag.Span}
-                  type={TypographyType.Callout}
-                  color={TypographyColor.Primary}
-                >
-                  {option.label}
-                </Typography>
```

Comment:
Looks to be a very big diff, but basically moved this to its own component file.